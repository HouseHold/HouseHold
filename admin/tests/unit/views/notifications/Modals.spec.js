import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import CoreuiVue from '@coreui/vue'
import Modals from '@/views/notifications/Modals'

Vue.use(CoreuiVue)

describe('Modals.vue', () => {
  it('has a name', () => {
    expect(Modals.name).toBe('Modals')
  })
  it('has a created hook', () => {
    expect(typeof Modals.data).toMatch('function')
  })
  it('sets the correct default data', () => {
    expect(typeof Modals.data).toMatch('function')
    const defaultData = Modals.data()
    expect(defaultData.largeModal).toBe(false)
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Modals)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Modals', () => {
    const wrapper = shallowMount(Modals)
    expect(wrapper.is(Modals)).toBe(true)
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Modals)
    expect(wrapper.element).toMatchSnapshot()
  })
})
