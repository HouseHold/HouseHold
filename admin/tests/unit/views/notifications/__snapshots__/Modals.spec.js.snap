// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modals.vue renders correctly 1`] = `
<div
  class="wrapper"
>
  <div>
    <crow-stub
      gutters="true"
      tag="div"
    >
      <ccol-stub
        col="12"
        tag="div"
      >
        <ccard-stub>
          <ccardheader-stub>
            <cicon-stub
              name="cil-justify-center"
            />
             Bootstrap Modals
            
            <div
              class="card-header-actions"
            >
              <a
                class="card-header-action"
                href="https://coreui.io/vue/docs/components/modal"
                rel="noreferrer noopener"
                target="_blank"
              >
                <small
                  class="text-muted"
                >
                  docs
                </small>
              </a>
            </div>
          </ccardheader-stub>
           
          <ccardbody-stub>
            <cbutton-stub
              activeclass="active"
              class="mr-1"
              color="secondary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              Launch demo modal
            
            </cbutton-stub>
             
            <cbutton-stub
              activeclass="active"
              class="mr-1"
              color="secondary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              Launch large modal
            
            </cbutton-stub>
             
            <cbutton-stub
              activeclass="active"
              class="mr-1"
              color="secondary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              Launch small modal
            
            </cbutton-stub>
             
            <hr />
             
            <cbutton-stub
              activeclass="active"
              class="mr-1"
              color="primary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              Launch primary modal
            
            </cbutton-stub>
             
            <cbutton-stub
              activeclass="active"
              class="mr-1"
              color="success"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              Launch success modal
            
            </cbutton-stub>
             
            <cbutton-stub
              activeclass="active"
              class="mr-1"
              color="warning"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              Launch warning modal
            
            </cbutton-stub>
             
            <cbutton-stub
              activeclass="active"
              class="mr-1"
              color="danger"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              Launch danger modal
            
            </cbutton-stub>
             
            <cbutton-stub
              activeclass="active"
              class="mr-1"
              color="info"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              Launch info modal
            
            </cbutton-stub>
             
            <cbutton-stub
              activeclass="active"
              class="mr-1"
              color="dark"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              Launch dark modal
            
            </cbutton-stub>
          </ccardbody-stub>
        </ccard-stub>
      </ccol-stub>
    </crow-stub>
  </div>
   
  <cmodal-stub
    backdrop="true"
    closeonbackdrop="true"
    fade="true"
    title="Modal title"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  
  </cmodal-stub>
   
  <cmodal-stub
    backdrop="true"
    closeonbackdrop="true"
    fade="true"
    size="lg"
    title="Modal title"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  
  </cmodal-stub>
   
  <cmodal-stub
    backdrop="true"
    closeonbackdrop="true"
    fade="true"
    size="sm"
    title="Modal title"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  
  </cmodal-stub>
   
  <cmodal-stub
    backdrop="true"
    closeonbackdrop="true"
    color="primary"
    fade="true"
    title="Modal title"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  
  </cmodal-stub>
   
  <cmodal-stub
    backdrop="true"
    closeonbackdrop="true"
    color="success"
    fade="true"
    title="Modal title"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  
  </cmodal-stub>
   
  <cmodal-stub
    backdrop="true"
    closeonbackdrop="true"
    color="warning"
    fade="true"
    title="Modal title"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  
  </cmodal-stub>
   
  <cmodal-stub
    backdrop="true"
    closeonbackdrop="true"
    color="danger"
    fade="true"
    title="Modal title"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  
  </cmodal-stub>
   
  <cmodal-stub
    backdrop="true"
    closeonbackdrop="true"
    color="info"
    fade="true"
    title="Modal title"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  
  </cmodal-stub>
   
  <cmodal-stub
    backdrop="true"
    centered="true"
    closeonbackdrop="true"
    color="dark"
    fade="true"
    no-close-on-backdrop="true"
    size="lg"
    title="Modal title 2"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    
  </cmodal-stub>
</div>
`;
